lib/maven/tools/model.rb

```diff
@@ -1,377 +1,292 @@
-require 'virtus'
+require 'dry-types'
+require 'dry-struct'
+require 'dry/struct/with_setters'
 
 # keep things in line with java collections
 class Array
   def remove( *args )
     delete( *args )
   end
 end
-    
-module Maven
-  module Tools
-    Base = Virtus.model
-
-    module GAV
-      def self.included( base )
-        base.attribute :group_id, String
-        base.attribute :artifact_id, String
-        base.attribute :version, String
-      end
-    end
-    module GA
-      def self.included( base )
-        base.attribute :group_id, String
-        base.attribute :artifact_id, String
-      end
-    end
-    module SU
-      def self.included( base )
-        base.attribute :system, String
-        base.attribute :url, String
-      end
-    end
-    module NU
-      def self.included( base )
-        base.attribute :name, String
-        base.attribute :url, String
-      end
-    end
-    module INU
-      def self.included( base )
-        base.attribute :id, String
-        base.attribute :name, String
-        base.attribute :url, String
-      end
-    end
-
-    class Parent
-      include Base
-
-      include GAV
```

Comment:
`include` only works with modules rather than classes, modules are vaguely analogous to  `interfaces` from Java.